src/visuals/sprite.rs

```diff
@@ -71,6 +87,12 @@ impl EventCollection {
     }
 }
 
+pub struct Animation {
```

Comment:
See above for comments.

Are you sure about using a `struct` for this? I'd be more willing to use an `enum`.

See my example https://gist.github.com/rust-play/7449903297b31642331a1da120667ae0#file-playground-rs-L7